src/extension/background-script/actions/nostr/generatePrivateKey.ts

```diff
@@ -0,0 +1,31 @@
+import * as secp256k1 from "@noble/secp256k1";
+
+import state from "../../state";
+
+const generatePrivateKey = async () => {
+  const connector = await state.getState().getConnector();
+  try {
+    const response = await connector.signMessage({
+      message:
+        "DO NOT EVER SIGN THIS TEXT WITH YOUR PRIVATE KEYS! IT IS ONLY USED FOR DERIVATION OF NOSTR KEY PAIRS, DISCLOSING THIS SIGNATURE WILL COMPROMISE YOUR NOSTR IDENTITY!",
+      key_loc: {
+        key_family: 0,
+        key_index: 0,
+      },
+    });
+    const keymaterial = response.data.signature;
```

Comment:
we probably should add some assertion here that we don't have a  blank signature.